projects/fsastorefrontlib/src/cms-components/my-account/policy/policies-chart/policies-chart.component.ts

```diff
@@ -0,0 +1,171 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { EChartsOption } from 'echarts';
+import { PolicyService } from '../../../../core/my-account/facade/policy.service';
+import { Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { PolicyChartDataService } from '../../../../core/my-account/services/policy-chart-data.service';
+
+@Component({
+  selector: 'cx-fs-policies-chart',
+  templateUrl: './policies-chart.component.html',
+})
+export class PoliciesChartComponent implements OnInit, OnDestroy {
```

Comment:
Maybe we can think about extracting some logic to one generic chart component which will have couple of input fields like chartOption, options, isChartVisibile etc.. such component can be used by policies, quotes, claims and other objects.. then PoliciesChartComponent can invoke new component and pass list of paymentFrequencies and some other policy relevant stuff for rendering...

Reply:
As we agreed I'll add task for improvement for this.